### ISSUE TITLE
[List] with hierarchy accessibility violation

### ISSUE THREAD

**Issue by yoshiemuranaka** [carbon-design-system/carbon-addons-iot-react#2094]
### What package is this for?

- [x] React
- [ ] Angular

### Describe the bug

The List with hierarchy has accessibility violations using the IBM Equal Access Accessibility Checker. The expand list icon does not have an accessible name. 

`Interactive component does not have a programmatically associated name`

### To Reproduce

Checked the list with hierarchy component in Storybook and ran the accessibility scan using the IBM Equal Access Accessibility Checker extension. Seeing the same issues while using the List component in my codebase 

### Environment/versions:

- OS: MacOS
- Browser: Chrome
- `carbon-addons-iot-react` version: v2.114.5

### Additional context

![Screen Shot 2021-03-05 at 10 13 48 AM](https://user-images.githubusercontent.com/8097306/110156214-8a61cc00-7d9b-11eb-869a-d202d7011fc7.png)

### Specific timeline issues / requests

Do you want this work within a specific time period? Is it related to an
upcoming release?

_NB: The core contributors will try to work with your timeline, but it's not
guaranteed. The earlier you make a request in advance of a desired delivery
date, the better!_
